azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -467,3 +552,73 @@ func resourceAzureRMKubernetesClusterServicePrincipalProfileHash(v interface{})
 
 	return hashcode.String(buf.String())
 }
+
+func base64Decode(str string) (string, bool) {
+	data, err := base64.StdEncoding.DecodeString(str)
+	if err != nil {
+		return "", true
+	}
+	return string(data), false
+}
+
+func getKubeConfig(config *string) *KubeConfig {
```

Comment:
_minor_ it might be worth pulling these out into a `kubernetes` specific package inside `./azurerm/helpers/kubernetes`?